tools/check_release_notes.sh

```diff
@@ -99,27 +111,38 @@ validate_opts() {
         print_error_and_exit "PULL_NUMBER is a required option. It must match the GitHub pull request number."
         exit 1
     fi
+
+    if [ -z "${REPO_PATH:-}" ]; then
+        echo "REPO_PATH not specified. Using current working directory."
+        REPO_PATH=$(pwd)
+    fi
+
+    if [ -z "${PULL_PULL_SHA:-}" ]; then
+        echo "PULL_PULL_SHA not specified. This must match the HEAD SHA for the pull request."
+        exit 1
+    fi
+
+    if [ -z "${PULL_BASE_REF:-}" ]; then
+        echo "PULL_BASE_REF not specified. This must match the target branch for the pull request."
+        exit 1
+    fi
 }
 
 # Curl the GitHub API to get a list of files for the specified PR. If files are
 # found, exit. We might eventually want to validate the data here.
 checkForFiles() {
-    echo "Requesting files from pull request ${REPO_OWNER}/${REPO_NAME}#${PULL_NUMBER}"
-    if [ -z "${token}" ]; then
-        ghFiles=$(curl -L -s --show-error --fail \
-            --header "Accept: application/vnd.github.v3+json" \
-            "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/pulls/${PULL_NUMBER}/files")
-    else
-        ghFiles=$(curl -L -s --show-error --fail \
-            --header "Accept: application/vnd.github.v3+json" \
-            --header "Authorization: Bearer ${token}" \
-            "https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/pulls/${PULL_NUMBER}/files")
-    fi
+    echo "Checking files from pull request ${REPO_OWNER}/${REPO_NAME}#${PULL_NUMBER} head SHA: ${PULL_PULL_SHA} destination branch: ${PULL_BASE_REF}"
+
+    pushd "${REPO_PATH}"
+
+    addedFiles=$(git diff "${PULL_BASE_REF}...${PULL_PULL_SHA}" --name-only --diff-filter=AMR)
```

Comment:
Does it work with three dots? In my system it's only two.

Reply:
sorry for the noise, you are right. I was testing on my system using wrong order of branches.